#ITSM-4047 dockerized yourl

sparshadotel opened the pull request:
- yourl and db running on separate containers

sparshadotel committed:
#ITSM-4047 dockerized yourl

cintiadr commented:
Alright, thanks for the PR :)

A lot of the comments come from:
https://github.com/openmrs/openmrs-contrib-ansible-docker-compose/blob/master/README.md

 
So the comments:
  - You do have access to the vault key in https://github.com/openmrs/openmrs-contrib-itsmresources/blob/master/ansible/.vault_pass (if you use git-crypt unlock). The readme there also mention how to use . So please go ahead and encrypt the .env with randomly generated passwords, and give a default value on docker-compose file (like the other files). 
  - Is there any reason why you cannot use sudoreboot/docker-yourls:latest or (even better image) bios/docker-yourls ? Do you have any reason why you'd fork and modify those images? The more public available code we use, tends to be the better. 
  - Can you configure yourls and mysql to have database, username and password using the same variables, instead of duplicating it?
  - Can you create a volume for mysql data and any other folder that yourls write in disk? (if they keep any state in disk). Also, you need to create the _deploy.env_ file if restarts shouldn't erase volumes. 
  - Can you please create healthchecks?
  - Can you add a restart:always?
  - Can you please not bind to port 80? We already have nginx on the host running there. It can be anything except 80 and 443. 
  - The docker image doesn't appear to be configuring port 443 (and it shouldn't, in fact). I suppose there's nothing being exposed on 443.
  - Can you add backups to mysql? https://github.com/openmrs/openmrs-contrib-ansible-docker-compose/blob/master/files/msf/docker-compose.yml#L41 . I don't really know if there are any other volumes/files you should keep, but it should be important. 

Tonight I will create a machine where you'll be able to deploy this.

sparshadotel committed:
Major Changes

- Added backup cron
- Encrypted .env with ansible vault
- Added default values to environment variables
- Added health checks
- Option to map volumes

sparshadotel commented:
- bios/docker-yourls kept redirecting me to https and sudoreboot/docker-yourls was not loading resources properly.
- daftlabs/docker-yourls repo had a problem with dockerfile variable so i had to fork it and fix it.
